external-health-monitor/release.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+# This file is part of MinIO DirectPV
+# Copyright (c) 2021, 2022, 2023 MinIO, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ME=$(basename "$0"); export ME
+cd "$(dirname "$0")" || exit 255
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+declare BUILD_VERSION PODMAN IMAGE_TAG_BASE IMG
+
+function init() {
+    if [ "$#" -ne 1 ]; then
+        cat <<EOF
+USAGE:
+  ${ME} <VERSION>
+
+EXAMPLE:
+  $ ${ME} 0.10.0
+EOF
+        exit 255
+    fi
+
+    BUILD_VERSION="${1/v/}"
+    IMAGE_TAG_BASE=quay.io/praveen8/csi-external-health-monitor-controller
+    IMG="${IMAGE_TAG_BASE}:v${BUILD_VERSION}"
+
+    if which podman >/dev/null 2>&1; then
+        PODMAN=podman
+    elif which docker >/dev/null 2>&1; then
+        PODMAN=docker
+    else
+        echo "no podman or docker found; please install"
+        exit 255
+    fi   
+}
+
+function main() {
+    # Downloading the source tar
+    curl --silent --location --insecure --fail "https://github.com/kubernetes-csi/external-health-monitor/archive/refs/tags/v${BUILD_VERSION}.tar.gz" | tar -zxf -
```

Comment:
Create a temporary build directory and do everything inside the directory. Cleanup the directory on exit.